### ISSUE TITLE
Add an example with `generic_arg_infer`

### ISSUE THREAD

**Issue by M1ngXU** [coreylowman/dfdx#119]
This is really useful to not specify the input size when the output size is clear (though this is not possible with type-aliases).
For example:
```rust
let model: (Linear<2, 5>, Linear<5, 2>) = Default::default();
model.forward(Tensor1D::ones().traced());
```
can be shortened with `#![feature(generic_arg_infer)]` to:
```rust
let model: (Linear<2, 5>, Linear<_, 2>) = Default::default();
model.forward(Tensor1D::ones().traced());
```

The second line is somehow necessary to infer the `_`.

**Comment by coreylowman** [comment]
Ooooh didn't know about that feature, great idea! Could probably even do that for the input size as well, and maybe specify input size in the input tensor? That would be SUPER useful for re-using a model.

Was not expecting to get the equivalent of "LazyLinear" for free this easily!

**Comment by M1ngXU** [comment]
> The second line is somehow necessary to infer the `_`.

The second line is obviously necessary since the model is just a non-related tuple xD

**Comment by M1ngXU** [comment]
> Was not expecting to get the equivalent of "LazyLinear" for free this easily!

What is “LazyLinear”?

**Comment by coreylowman** [comment]
Basically pytorch version of various values being inferred at runtime

https://pytorch.org/docs/stable/generated/torch.nn.LazyLinear.html

**Comment by M1ngXU** [comment]
Already closed with #247
